tests/generic_config_updater/test_dynamic_acl.py

```diff
@@ -0,0 +1,1181 @@
+        }
+        patch = {
+            "op": "add",
+            "path": full_rule_name,
+            "value": rule_vals
+        }
+        json_patch.append(patch)
+        expected_content = ["DYNAMIC_ACL_TABLE", rule_name, str(priority), "DROP", "IN_PORTS: " + port_name, "Active"]
+        expected_rule_contents[rule_name] = expected_content
+        priority -= 1
+        rule_number += 1
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
+
+    try:
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_success(duthost, output)
+
+        for rule_name, expected_content in expected_rule_contents.items():
+            expect_acl_rule_match(duthost, rule_name, expected_content)
+
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+
+def dynamic_acl_remove_ip_forward_rule(duthost, ip_type):
+    """Remove selected forward rule from the acl table"""
+
+    if ip_type == "IPV4":
+        rule_name = "RULE_1"
+    else:
+        rule_name = "RULE_2"
+
+    extra_vars = {
+        "rule_name": rule_name
+    }
+
+    output = format_and_apply_template(duthost, REMOVE_RULE_TEMPLATE, extra_vars)
+
+    expect_op_success(duthost, output)
+
+    expect_acl_rule_removed(duthost, rule_name)
+
+
+def dynamic_acl_remove_table(duthost):
+    """Remove an ACL Table Type from the duthost"""
+
+    output = load_and_apply_json_patch(duthost, REMOVE_TABLE_FILE)
+
+    expect_op_success(duthost, output)
+
+
+def dynamic_acl_remove_nonexistent_table(duthost):
+    """Remove a nonexistent ACL Table from the duthost, verify it fails"""
+
+    output = load_and_apply_json_patch(duthost, REMOVE_NONEXISTENT_TABLE_FILE)
+
+    expect_op_failure(output)
+
+
+def dynamic_acl_remove_table_type(duthost):
+    """Remove an ACL Table definition from the duthost"""
+
+    output = load_and_apply_json_patch(duthost, REMOVE_TABLE_TYPE_FILE)
+
+    expect_op_success(duthost, output)
+
+
+def test_gcu_acl_arp_rule_creation(rand_selected_dut,
+                                   ptfadapter,
+                                   setup,
+                                   dynamic_acl_create_table,
+                                   prepare_ptf_intf_and_ip,
+                                   toggle_all_simulator_ports_to_rand_selected_tor):  # noqa F811
+    """Test that we can create a blanket ARP packet forwarding rule with GCU, and that ARP packets
+    are correctly forwarded while all others are dropped.  IPv6 NDP testing not supported at this time."""
+
+    ptf_intf_ipv4_addr, ptf_intf_ipv6_addr, _, ptf_intf_index, port_name = prepare_ptf_intf_and_ip
+
+    dynamic_acl_create_arp_forward_rule(rand_selected_dut)
+    dynamic_acl_create_secondary_drop_rule(rand_selected_dut, setup, port_name)
+
+    rand_selected_dut.shell("sonic-clear arp")
+    # give command time to finish
+    time.sleep(10)
+    rand_selected_dut.shell("ping -c 3 {}".format(ptf_intf_ipv4_addr))
```

Comment:
1. The ping command should fail here because the ICMP packet is dropped by the dropping rule. So we need to ignore the error for the ping command.
2. Please clear ARP after removing the IP address from the ptf port (can be done in the fixture).
3. Seems IPv6 neighbor learning is missed here.

Reply:
Manual testing passed, so I believe the error is already being ignored with current code setup?  I believe in other tests to see that a shell command failed you would need to set `output = rand_selected_dut.shell(....)`, and then check the value of `output['rc']`.  As we are not performing that check, I believe that even though the duthost may return an error, it does not effect the test.  Let me know if I am mistaken here.

I will clear the ARP from within the end of the fixture.

IPv6 neighbor learning is missed by design, as NDP forwarding rules cannot be added at this time (unless I have missed a SONiC buildimage update that would include your fix for ICMP ranges)